src/olm/account/mod.rs

```diff
@@ -439,6 +444,79 @@ impl Account {
 
         pickle.try_into()
     }
+
+    /// Create a dehydrated device from the account.
+    ///
+    /// A dehydrated device is a device that is stored encrypted on the server
+    /// that can receive messages when the user has no other active devices.
+    /// Upon login, the user can rehydrate the device (using
+    /// [`from_dehydrated_device`]) and decrypt the messages sent to the
+    /// dehydrated device.
+    ///
+    /// Returns the ciphertext and nonce.  `key` is a 256-bit (32-byte) key for
+    /// encrypting the device.
+    ///
+    /// The format used here is defined in
+    /// [MSC3814](https://github.com/matrix-org/matrix-spec-proposals/pull/3814).
+    pub fn to_dehydrated_device(
+        &self,
+        key: &[u8; 32],
+    ) -> Result<(String, String), crate::DehydratedDeviceError> {
```

Comment:
Should I create a new struct instead of returning `(String, String)`?

Reply:
Yeah that sounds good, and it might implement `Serialize`